superset/db_engine_specs/base.py

```diff
@@ -161,7 +138,7 @@ def get_timestamp_expr(
         :return: TimestampExpression object
         """
         if time_grain:
-            time_expr = cls.time_grain_functions.get(time_grain)
+            time_expr = cls.get_time_grain_functions().get(time_grain)
```

Comment:
[optional] we may want to materialize the result of `get_time_grain_functions()` instead of re-computing it every time. This being a static class (all @classmethods) it's a bit funky to do (no constructor). Maybe the `@memoize` decorator we have around somewhere would work.

May be overkill as this method may not be called much...

Reply:
I agree; while this should be minimal overhead, stuff like this tends to cause problems down the line left untreated. To avoid the risk of introducing new regressions I will not refactor this now, but added a `TODO` that I will try to remove in the near future (will aim to kill this when removing `name` from `TimeGrain`).